internal/factsengine/gathering.go

```diff
@@ -24,7 +26,7 @@ func gatherFacts(
 	}
 	groupedFactsRequest := groupFactsRequestByGatherer(agentFacts)
 	factsCh := make(chan []entities.Fact, len(groupedFactsRequest.FactRequests))
-	g := new(errgroup.Group)
+	g, groupCtx := errgroup.WithContext(context.Background())
```

Comment:
We should get this context still from above.
The context we use everywhere should be the one coming in the start of the agent (`a.Start(ctx)`) in the `start.go` file.
Otherwise, we don't use the same context.
I know that we need to pass this context in many many functions, but I think it is worthy. Just put `ctx context.Context` in a bunch of functions.
What do you think?

Reply:
I'm not sure the context to use is the one relative to the `Start` command. This context is scoped on each gathering request (coming from rabbitmq).

As I intended it, we want all routines in the same request to terminate on failure. If we use the root context, we will have the whole agent process to fail.

Reply:
> This context is scoped on each gathering request (coming from rabbitmq).

And it would continue this way, right? We would just be creating a new grouped context that can still be cancelled from above.

> If we use the root context, we will have the whole agent process to fail.
Is this true? 
I wouldn't expect the parent context being cancelled if some process in the group fails.
Maybe I'm wrong.

But what I would want is that if the parent context is cancelled (we stop trento agent), the gathering is cancelled and the execution stops, without leaving open traces or processes running.

If we create a new empty context, the stopping of the agent is not propagated to the gathering, and the gatherers that use contexts neither.

Reply:
You're right, I didn't consider that scenario. I'll come up with a solution. Thanks!